lua_cjson.c

```diff
@@ -1215,6 +1231,16 @@ static void json_parse_array_context(lua_State *l, json_parse_t *json)
 
     /* Handle empty arrays */
     if (token.type == T_ARR_END) {
+        // Mark as array:
+        luaL_getmetatable(l, "cjson.array");
+        if ( lua_isnil(l, -1) ) {
+            lua_pop(l, 1);
+            luaL_newmetatable(l, "cjson.array");
```

Comment:
Because this metatable is readonly. Could we share it globally?

Also, I hope that tagging empty arrays is optional so that it does not change the current default behaviour and also we don't have to pay this extra overhead if we don't want to serialise it again. Thoughts?